File: DS/api/java/CreateScheduledTask.java
import.com.trendmicro.deepsecurity.ApiClient;

/* class is an example of how to create a scheduled task to run recommendation scan for all computers in Deep Security
 * Manager using APIs
 */
 
 public class CreateScheduledTask {
 
    public static void main(String[] args) {
    //setup the connection to Deep Security manager
    // You need to enter your DSM URL
    ApiClient defaultClient = Configuration.getDefaultApi();
    // BasePath needs to be https://<your DSM FQDN or IP>:4119/api
    defaultClient.setBasePath("<Your Hostname>:4119/api");
    
    // Authentication
    // You need to enter your API Key
    ApiKeyAuth DefaultAuthentication = (ApiKeyAuth) defaultCilent.getAuthenication("DefaultAuthentication");
    DefaultAuthentication.setApiKey ("<Your API Key>");
    try {
       defaultClient.trustAllCertificates(false);
    } catch (Exception e) {
       System.err.println ("An exception occured when calling ApiClient.trustAllCertificates");
       e.printStackTrace();  
    }
      
    // Initialization
    // Here the new scheduled task will be setup up
    ScheduledTasksApi instance = new ScheduledTasksApi();
    ScheduledTask scheduledTask  = new ScheduledTask();
    // Set scheduled task name
    scheduledtask.setName ("API Created - Scheduled Recommenation Scan Task");
    // Set up the details - time when the task will run 
    ScheduleDetails scheduleDetails = new ScheduledDetails();
    scheduleDetails.setTimeZone("Europe/Berlin");
    scheduleDetails.setRecurrenceType(RecurrenceTypeEnum.WEEKLY);
    // Set the weekly parameters
    WeeklyScheduledParameters weeklyScheduleParameters = new WeeklyScheduleParameters();
    // set the schedule time , i.e. for start at 12/16/2018 at 1 AM
    String startDate = "2019-12-16 01:00:00";
    SimpleDateFormat dateFormatter = new SimpleDateFormat ("yyyy-MM-dd HH:mm:ss");
    try {
       Date parseDate = dateFormatter.parse(startDate);
       long startTime = parseDate.getTime();
       weeklyScheduleParameters.startTime(startTime);
    } catch (Exception e) {
       e.printStackTrace();
    } 
    // Set the task to run every 1 wwek
    // Set it for all computers
    ScanForRecommendationsTaskParameters scanForRecommndationTaskParameters = new ScanForRecommendationsTaskParameters();
    ComputerFilter computerFilter - new ComputerFilter();
    computerFilter.setType(ComputerFilter.TypeEnum.ALL_COMPUTERS);
    scanForRecommendationTaskParameters.setComputerFilter(computerFilter);
    scheduledTask.setScanForRecommendationTaskParameters(scanForRecommendationTaskParameters);
        
    String apiVersion = "v1";
    try {
       // create the new scheduled scan for recommenation task
       ScheduledTask result = instance.createScheduledTask(scheduledTask, apiVersion);
       system.out.println(result);
    } catch (ApiException e) {
       System.err.println("An exception occured when calling ScheduledTaskApi.createAdministrator");
       e.printStackTrace();
    } 
 }

File: DS/api/python/create_scheduled_task.py
from __future__ import print_function
import sys, warnings
import deepsecurity
from deepsecurity.rest import ApiException
from print import pprint

# Setup
if not sys.warnoptions:
   warnings.simplefilter("ignore")
configuration = deepsecurity.Configuration()
# host must be set to 
configuration.host = '<Your Hostname or IP>'

# Authentication
# The API key must be set to
configuration.api_key['api-secret-key'] = '<Your API Key>'

# Initialization
# Set any required values
api_instance = deepsecurity.ScheduledTaskApi (deepsecurity.ApiClient(configuration))
scheduledTask = deepsecurity.ScheduledTask()
api_version = 'v1'

# set the task name
scheduledTask.name = "APi Created Scheduld Recommendation Scan Task"
# set the task  type
scheduledTask.type = "scan-for-recommendation"
# set the task details like time to run
scheduledTaskParameter = deepsecurity.SchedleDetails()
scheduledTaskParameter.time_zone = "Europe/Berlin"
scheduledTaskParameter.recurrence_type = "weekly"
# set the task weekly run 
weeklyParameters =  deepsecurity.weeklyScheduleParameters()
weeklyParameters.interval = "1"
weeklyParameters.days =  "sunday"
weeklyParameters = 1544922000000
# set the scheduled task details with all the information
scheduledTaskParaemtrs.weekly_schedule_parameters = weeklyParameters
scheduled_task.schedule_details = scheduledTaskParameters
# set the recommendation task information
# run for all computers
recommendationScanParameters = deepsecurity.ScanForRecommendationsTaskParamters()
computerFilter = deepsecurity.CompuerFilter()
computerFilter.type = "all-computers"
recommendationsScanParameters.computer_filter = computerFilter
# set the scheduled task details
scheduled_task.scan_for_recommendation_task_parameters = recommendationScanParameters

try: 
   api_response = api_instance.create_scheduled_task(scheduled_task, api_version, api_version)
   pprint(api_response)
except ApiException as e:
   print ("An exception occured when calling ScheduledTasksApi.create_scheduled_task: %s\n" % e)
